src/seq_mv/vector_device.c

```diff
@@ -151,6 +151,48 @@ hypre_SeqVectorAxpyDevice( HYPRE_Complex alpha,
    return hypre_error_flag;
 }
 
+/*--------------------------------------------------------------------------
+ * hypre_SeqVectorAxpyzDevice
+ *--------------------------------------------------------------------------*/
+
+HYPRE_Int
+hypre_SeqVectorAxpyzDevice( HYPRE_Complex  alpha,
+                            hypre_Vector  *x,
+                            HYPRE_Complex  beta,
+                            hypre_Vector  *y,
+                            hypre_Vector  *z )
+{
+   HYPRE_Complex  *x_data      = hypre_VectorData(x);
+   HYPRE_Complex  *y_data      = hypre_VectorData(y);
+   HYPRE_Complex  *z_data      = hypre_VectorData(z);
+
+   HYPRE_Int       num_vectors = hypre_VectorNumVectors(x);
+   HYPRE_Int       size        = hypre_VectorSize(x);
+   HYPRE_Int       total_size  = size * num_vectors;
+
+#if defined(HYPRE_USING_CUDA) || defined(HYPRE_USING_HIP)
+   hypreDevice_ComplexAxpyzn(total_size, x_data, y_data, z_data, alpha, beta);
+
+#elif defined(HYPRE_USING_SYCL)
+   HYPRE_ONEDPL_CALL( std::transform,
+                      x_data, x_data + total_size, y_data, z_data,
+                      [alpha, beta](HYPRE_Complex x, HYPRE_Complex y) -> HYPRE_Complex { return alpha * x + beta * y; } );
```

Comment:
@waynemitchell Does this make sense?

Reply:
@victorapm , this looks correct and compiles for me on JLSE, but I'm curious why you make a special case for sycl here? The call to hypreDevice_ComplexAxpyzn() above will also work for sycl, so I think it's better to have all backends (cuda/hip/sycl) call the same code for simplicity.

Reply:
Oh yes! You are correct, it's definitely better to only have `hypreDevice_ComplexAxpyzn` (which by the way needs a name update). Not sure why I put that onedpl call there. I'll remove it, thanks!!